src/test/ui/const-generics/str-const-param.rs

```diff
@@ -0,0 +1,12 @@
+// run-pass
```

Comment:
Would be good to have a failing test for the `Err(TypeError::ConstMismatch(expected_found(relation, &a, &b)))` as well.